### PR TITLE
Add MedNeXt-L + SkeletonRecall trainer for compressed surface regions

### PR DESCRIPTION
**Pull request by ciscoriordan**
## What

Adds `nnUNetTrainerSkeletonRecall_MedNeXtL_kernel5` (plus a kernel3 fallback in the same file) to `segmentation/models/arch/nnunet/nnunetv2/training/nnUNetTrainer/variants/loss/`. The trainer extends the existing `nnUNetTrainerSkeletonRecall` and only overrides `build_network_architecture` to swap the default ResEnc U-Net for MedNeXt-L. Loss, transforms, data loaders, and train/validation steps are inherited unchanged.

## Why

Addresses #191 ("Surface and Fiber Predictions in Compressed or Highly Curved areas"). The d058 ResEnc-L production model under-predicts in compressed / highly curved cubes. MedNeXt-L's depthwise-then-pointwise inverted-bottleneck blocks (the "ConvNeXt for medical seg" recipe) give a noticeably better fit there while keeping the rest of bruniss's pipeline (SkeletonRecall loss, augmentations, Dataset059 layout) intact.

On a held-out 5-cube benchmark over PHerc Paris 1 (S1) and PHerc 1667 (S4):

| metric | d058 (ResEnc-L) | MedNeXt-L k5 + SkelRec (this) | delta |
|---|---|---|---|
| high_compressed IoU | 0.404 | 0.671 | +0.267 (+66%) |
| overall_macro IoU | 0.534 | 0.685 | +0.151 |

Full methodology, per-cube numbers, and the comparison vs. UMambaEnc are in [the issue comment](https://github.com/ScrollPrize/villa/issues/191#issuecomment-4472801908).

## How to use

```
pip install -e "segmentation/models/arch/nnunet[mednext]"
nnUNetv2_train DATASET_ID 3d_fullres 0 -tr nnUNetTrainerSkeletonRecall_MedNeXtL_kernel5
```

For VRAM-tight setups, swap in `nnUNetTrainerSkeletonRecall_MedNeXtL_kernel3` (same file).

Pretrained weights on Hugging Face (best checkpoint `kernel5_skelrec_dataset059_ep33/`):
https://huggingface.co/ciscoriordan/mednext-l-scroll-surface

## What's NOT included

- No MedNeXt source vendored. The trainer imports from the upstream `mednextv1` pip package (https://github.com/MIC-DKFZ/MedNeXt), added as an optional dependency under `[project.optional-dependencies].mednext` so users not using this trainer see no install change.
- No preset training plans / batch_running configs. Out of the box this uses the same plans the existing `nnUNetTrainerSkeletonRecall` runs with (e.g. `nnUNetResEncUNetMPlans`); MedNeXt's `build_network_architecture` ignores the ResEnc-specific architecture kwargs and just uses MedNeXt's own block schedule.
- No changes to `nnUNetTrainerSkeletonRecall` or any other existing trainer.
- No CI / build configuration changes.

## Files changed

- `segmentation/models/arch/nnunet/nnunetv2/training/nnUNetTrainer/variants/loss/nnUNetTrainerSkeletonRecall_MedNeXtL_kernel5.py` (new, ~95 lines including docstring)
- `segmentation/models/arch/nnunet/pyproject.toml` (added optional `mednext` extra)
- `segmentation/models/arch/nnunet/README.md` (short section under existing `### train` block, pointing to #191, the benchmark numbers, HF weights, and the kernel3 fallback)

## Open for discussion

- File location: I placed the trainer in `variants/loss/` next to its parent `nnUNetTrainerSkeletonRecall.py`, since it's a near-zero-glue subclass of that trainer (loss is the dominant axis). The alternative would be `variants/network_architecture/`. Happy to move it if maintainers prefer.
- Marking as draft so maintainers can flag placement / naming / extras-grouping preferences before merge.

Source repo (public) with the original training scripts and benchmark code: https://github.com/ciscoriordan/mednext-vs-umamba-scroll

### CODE REVIEW

**Comment by bruniss** [comment]
image

i'm not sure how to interpret this image other than that i think the task here was misunderstood. the image you have shown and purported to have "high mergers" is in fact perfectly separated and contains no mergers, while the proposed solution has merged every single sheet together. 

if you are interested in doing ml based recto surface segmentation i would suggest starting with reviewing the kaggle competition and some of the discussion there.

**Comment by bruniss** [comment]
here is a link to the kaggle competition i forgot to include in my previous comment https://www.kaggle.com/competitions/vesuvius-challenge-surface-detection
